nautobot/docs/development/core/model-checklist.md

```diff
@@ -0,0 +1,139 @@
+# Model Development Checklist
+
+A general best-practices checklist to follow when adding a net-new data model in Nautobot or adding new fields to an existing model.
+
+## Bootstrapping a new Model
+
+- ▢ Implement model in `nautobot.<app>.models` module
+    - ▢ Use appropriate base class (`PrimaryModel` or `BaseModel`) and mixin(s)
+    - ▢ Use appropriate `@extras_features` decorator values
+    - ▢ Define appropriate uniqueness constraint(s)
+    - ▢ Define appropriate `__str__()` logic
+    - ▢ _optional_ Define appropriate additional `clean()` logic
+        - ▢ Be sure to always call `super().clean()` as well!
+    - ▢ _optional_ Define appropriate `verbose_name`/`verbose_name_plural` if defaults aren't optimal
+- ▢ Generate database schema migration(s) with `invoke makemigrations <app> -n <migration_name>`
+- ▢ Implement API `<Model>Serializer` class in `nautobot.<app>.api.serializers` module
+    - ▢ Use appropriate base class and mixin(s)
+- ▢ Implement `<Model>FilterSet` class in `nautobot.<app>.filters` module
+    - ▢ Use appropriate base class (`NautobotFilterSet` or `BaseFilterSet`) and mixin(s)
```

Comment:
`best-practices.md` says this:

> FilterSets must inherit from nautobot.extras.filters.NautobotFilterSet (which inherits from nautobot.core.filters.BaseFilterSet)

Reply:
Updating that document - there are cases for `BaseFilterSet` for certain models.